pydatastructs/graphs/algorithms.py

```diff
@@ -762,6 +767,40 @@ def _dijkstra_adjacency_list(graph: Graph, start: str, target: str):
 
 _dijkstra_adjacency_matrix = _dijkstra_adjacency_list
 
+def _floyd_warshall_adjacency_list(graph: Graph, source: str, target: str):
+    dist, predecessor, dist_temp, pred_temp = dict(), dict(), dict(), dict()
```

Comment:
I am unable to understand the purpose of `dist_temp`, `pred_temp`. In L778 to L780 if they are being copied to `dist[v]` and `predecessor[v]`. May be, we could have directly followed, https://en.wikipedia.org/wiki/Floyd%E2%80%93Warshall_algorithm#Pseudocode_[11] word to word.